### PR TITLE
refactor: Rewrite convertImmutable

### LINKED ISSUE #2444: Rewrite convertImmutable
It is too horrible, and quick to do.

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #2444

## Description

Rewrites convertImmutable.

### CODE REVIEW

**Review by shahzadlone — APPROVED** [review]
LOL at issue description `It is too horrible` 😆 

LGTM

File: client/value.go [hunk 1]
```diff
@@ -79,13 +79,11 @@ func (val FieldValue) Bytes() ([]byte, error) {
...
 func convertImmutable[T any](vals []immutable.Option[T]) []any {
-	var out []any
-	for _, val := range vals {
-		if !val.HasValue() {
-			out = append(out, nil)
-			continue
+	out := make([]any, len(vals))
+	for i := range vals {
+		if vals[i].HasValue() {
+			out[i] = vals[i].Value()
```
Comment (islamaliev) [inline client/value.go:85]: suggestion: this is a small change with potentially significant impact. I wonder why no tests cover this.
Would be great to add some tests.

**Review by islamaliev — APPROVED** [review]

File: client/value.go (hunk 1 above)
Comment (AndrewSisley) [reply client/value.go:85]: It is tested, I was mistaken in your PR - I just rewrote it because it was really ugly

Comment (fredcarle) [reply client/value.go:85]: I don't think this change has any impact other than improving the efficiency of the code. The returned result is the same.

**Pull request merged**